### ISSUE TITLE
Campaign builder page: willing party

### ISSUE THREAD

**Issue by tdg** [swapmyvote/swapmyvote#643]
re #639
Ability to specify a party and return the name of that party as a slug to the URL builder willing party part of the campaign builder page

**Comment by baob** [comment]
We need to consider the answer to 
- #489 

to do this. We can't go distributing permanent URLs including party slugs when we have no fixed definition of what the party slugs are.

**Comment by tdg** [comment]
> We need to consider the answer to
> 
> * [Decide selection criteria for 'all parties' at next GE #489](https://github.com/swapmyvote/swapmyvote/issues/489)
> 
> to do this. We can't go distributing permanent URLs including party slugs when we have no fixed definition of what the party slugs are.

Agree - good spot - but we don't need to do that in order to meet this issue, do we? Especially for an MVP.
We would have to do it before we publish and promote the campaigner site though,  and that consideration would need to include planning how to deal with changes to party names etc, anything that could make a 'permanent' link impermanent.

So I'd say there's a difference between #489 being a blocker to progress, and a blocker to distribution of the Campaigner tool (also for #645)

**Comment by baob** [comment]
> So I'd say there's a difference between #489 being a blocker to progress, and a blocker to distribution of the Campaigner tool (also for #645)

I think I'd agree with that.

It's worth noting, though, that there's no such thing as 'permanent' party slug which then changes/becomes impermanent. They change with every election (or rather, there's no guarantee they will stay the same, and they have definitely changed in the past). They only become permanent once we freeze the number of parties we're considering for the election.

**Comment by tdg** [comment]
Yes good point. In fact we might well consider taking a leaf out of Democracy Club's book and specifying the election we're serving as part of the URL (not just for  API calls but also got the site itself?!)

**Comment by aspiers** [comment]
I don't see a pressing need for permanent slugs or campaign URLs.  It seems quite unlikely that any significant number of people would try to reuse the same links across elections.  Anyway we could easily add validation which catches the case where invalid URL parameters are used, and then just show an error and nudge the user back to the URL builder page in those cases.

So I'm pretty sure that #489 is not a blocker in any way.

Consequently, adding the election as part of the URL seems an unnecessary complication which takes us in the opposite direction to the URL simplification / shortening which we wanted.
